### PR TITLE
Enhancement charity location

### PR DESCRIPTION
**Pull request by baikauskaite**
Made charity addresses viewable. You can also select the address and view the location on the map.
You have to use the API key to see the map (without it there will just be no map shown). 

:arrow_down: For more information on how to set up, see down below. 

Screenshot 2021-12-04 at 12 43 03

### Steps to do before reviewing:
1. `update-database`, like always
2. Go to `Project` > `Manage user secrets`, this will create a `secrets.json` file. Paste the API key there. Please contact me to get the API key.
3. You can change the addresses in `CharityAddresses` table to more accurate addresses to test the legitimacy of the map.

Screenshot 2021-12-04 at 12 01 07

Once you will set up the API key you will never have to do that again.

### The logic behind this:
- The longitude and latitude are counted for the given address and in that way the location can be shown on the map.
- If the address is inaccurate, the location that is nearest by is shown on the map.

### Other changes made:
- Separated the charity details used in the CharityManager and details shown in Charities' search. Now there are 2 views: `Details.cshtml` and `DetailsForCharityManager.cshtml`

Note: first 2 commits are from my last PR.

### CODE REVIEW

**Review by PauliusLu — COMMENTED** [review]
Code looks good, did everything written in PR description, but throws an error for me.
image

**Comment by baikauskaite** [comment]
> Code looks good, did everything written in PR description, but throws an error for me.
> image

Thank you for the comment, though I'll need more debugging on your machines to figure out what's the problem.

**Review by ArmintasP — COMMENTED** [review]
I didn't encounter same errors as @PauliusLu did, everything worked fine on my browsers (Firefox94, Chrome96).

File: Karma/Startup.cs [hunk 1]
```diff
@@ -57,6 +57,11 @@ public void ConfigureServices(IServiceCollection services)
             services.AddDbContext<KarmaContext>(options =>
                     options.UseSqlite(Configuration.GetConnectionString("DefaultConnection")));
 
+            // Adding singleton for Google Maps API
+            var maps = new GoogleMaps();
+            maps.ServiceApiKey = Configuration.GetValue<string>("GoogleMaps:ServiceApiKey");
+            services.AddSingleton<GoogleMaps>(maps);
```
[the remark this answers is not in the document]
Comment (baikauskaite) [reply Karma/Startup.cs:63]: Good idea, will do that.

**Pull request merged**